src/tests/ftest/osa/osa_offline_reintegration.py

```diff
@@ -24,111 +25,195 @@ def setUp(self):
         """Set up for test case."""
         super(OSAOfflineReintegration, self).setUp()
         self.dmg_command = self.get_dmg_command()
-        self.ior_apis = self.params.get("ior_api", '/run/ior/iorflags/*')
-        self.ior_test_sequence = self.params.get(
-            "ior_test_sequence", '/run/ior/iorflags/*')
-        self.ior_dfs_oclass = self.params.get(
-            "obj_class", '/run/ior/iorflags/*')
+        self.daos_command = DaosCommand(self.bin)
+        self.ior_test_sequence = self.params.get("ior_test_sequence",
+                                                 '/run/ior/iorflags/*')
+        self.test_oclass = self.params.get("oclass", '/run/test_obj_class/*')
+        self.loop_test_cnt = 1
         # Recreate the client hostfile without slots defined
         self.hostfile_clients = write_host_file(
             self.hostlist_clients, self.workdir, None)
+        self.dmg_command.exit_status_exception = True
 
     def run_offline_reintegration_test(self, num_pool, data=False,
-                                       server_boot=False):
+                                       server_boot=False, oclass=None,
+                                       reint_during_rebuild=False,
+                                       reint_during_aggregation=False):
         """Run the offline reintegration without data.
             Args:
             num_pool (int) : total pools to create for testing purposes.
             data (bool) : whether pool has no data or to create
                           some data in pool. Defaults to False.
             server_boot (bool) : Perform system stop/start on a rank.
                                  Defults to False.
+            oclass (str) : daos object class string (eg: "RP_2G8")
+            reint_during_rebuild (bool) : Perform reintegration during
+                                          rebuild (Defaults to False).
+            reint_during_aggregation (bool) : Perform reintegration
+                                              during aggregation
+                                              (Defaults to False).
         """
         # Create a pool
         pool = {}
-        pool_uuid = []
-        exclude_servers = (len(self.hostlist_servers) * 2) - 1
-
-        # Exclude rank : two ranks other than rank 0.
-        rank = random.randint(1, exclude_servers)
+        random_pool = 0
+        if oclass is None:
+            oclass = self.ior_cmd.dfs_oclass.value
 
+        # Exclude ranks [0, 3, 4]
+        rank = [0, 3, 4]
```

Comment:
Can we define the ranks to be excluded/integrated/drained/reintegrated from yaml.

Reply:
We should add this in yaml. May be in future PRs.